### PR TITLE
added statement in register method

### PR DESCRIPTION
**Pull request by Nonzeq**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory()
+                    .openSession();
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java:21]:
```suggestion
            session = HibernateUtil.getSessionFactory().openSession();
```

Comment (Sanderfuck) [reply src/main/java/mate/academy/dao/impl/UserDaoImpl.java:21]: Your replacement of code doesn`t make any profit

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory()
+                .openSession()) {
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java:41]:
```suggestion
        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Review by Sanderfuck — APPROVED** [review]

**Review by Sanderfuck — APPROVED** [review]

File: src/main/java/mate/academy/model/User.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String email;
```
Comment (Rommelua) [inline src/main/java/mate/academy/model/User.java:15]: image

**Pull request closed without merging**